### PR TITLE
Nawqaan 577 client side sos shim

### PR DESCRIPTION
**Pull request by cschroed-usgs**

### CODE REVIEW

**Comment by cschroed-usgs** [comment]
These changes only make the site summary graphs ([example](http://cida.usgs.gov/quality/rivers/site/07288650/summarygraphs)) work. The detailed graphs page needs some more updates to work.

File: nar_common/static/nar_ui/time_series/TimeSeries.js [hunk 1]
```diff
@@ -22,86 +22,49 @@ nar.timeSeries.TimeSeries = function(config){
...
-                    var tokens = row.split(rowSplitToken);
-                    var timeStamp = Date.create(tokens[dateIndex]).getTime();
-                    //overwrite
-                    tokens[dateIndex] = timeStamp;
-                    return tokens;
-                });
-            }
+      
+    self.parseResultRetrievalResponse = function(response){
+        var dataToReturn;
```
Comment (jiwalker-usgs) [inline nar_common/static/nar_ui/time_series/TimeSeries.js]: Are you meaning to define dataToReturn twice?

**Comment by jiwalker-usgs** [comment]
Cool, seems a reasonable way to go to me.  How are you feeling about it?

:+1:

File: nar_common/static/nar_ui/util.js [hunk 2]
```diff
@@ -86,4 +95,119 @@ nar.util = {};
...
+	var sosProcedureToValueProperties = {
+			'annual_mass' : ['tons'],
+			'annual_concentration_flow_weighted' : ['fws'],
+			'monthly_mass' : ['tons'],
+			'monthly_flow' : ['flow'],
+			'daily_flow' : ['flow'],
+			'annual_flow' : ['flow'],
+			'discrete_concentration' : ['remark', 'concentration']
+	};
+
```
Comment (cschroed-usgs) [inline nar_common/static/nar_ui/util.js]: I haven't implemented tests for getValueForResponseRow. Should I?

File: nar_common/static/nar_ui/util.js [hunk 3]
```diff
@@ -86,4 +95,119 @@ nar.util = {};
...
+		var valueProperties = sosProcedureToValueProperties[procedure];
+		//value can be comprised of multiple fields. Join fields together in order.
+		var value = valueProperties.reduce(function(accumulation, current){
+			//if attribute is missing, use blank string
+			return accumulation + '' + (responseRow[current] || '');
+		}, '');
+		
+		return value;
+	};
+
```
Comment (cschroed-usgs) [inline nar_common/static/nar_ui/util.js]: Similarly, I haven't implemented tests for getTimestampForResponseRow. Should I?

File: nar_common/static/nar_ui/util.js [hunk 4]
```diff
@@ -86,4 +95,119 @@ nar.util = {};
...
+	
+	nar.util.getTimestampForResponseRow = function(responseRow){
+		var theDate;
+		if(responseRow.date){
+			theDate = Date.create(responseRow.date);
+		} else {
+			if(responseRow.month) {
+				theDate = Date.create(responseRow.wy + '-' + responseRow.month);
+			} else {
+				theDate = Date.create('' + responseRow.wy);
```
Comment (cschroed-usgs) [inline nar_common/static/nar_ui/util.js]: The absence of this string cast made things wonky for quite a while. I think I should probably test for that.

File: nar_common/static/nar_ui/util.js [hunk 5]
```diff
@@ -86,4 +95,119 @@ nar.util = {};
 		return hash;
 	};
```
Comment (cschroed-usgs) [inline nar_common/static/nar_ui/util.js]: Is it confusing for these shim functions be in the utils class? Would it be better to separate them into a SosShim file?

File: nar_common/static/nar_ui/time_series/TimeSeries.js (hunk 1 above)
Comment (cschroed-usgs) [inline nar_common/static/nar_ui/time_series/TimeSeries.js]: Nope! Good catch. Thanks.

**Comment by cschroed-usgs** [comment]
Thanks JW. I'm feeling okish about it. I'm curious to hear what you think on the questions I just added.

File: nar_common/static/nar_ui/util.js (hunk 5 above)
Comment (jiwalker-usgs) [inline nar_common/static/nar_ui/util.js]: I was thinking about that too, but I think they are fine here.  The argument to pull them out is that they are used together and aren't floating utils that are independent.

File: nar_common/static/nar_ui/util.js (hunk 4 above)
Comment (jiwalker-usgs) [inline nar_common/static/nar_ui/util.js]: Yeah, a test for this would definitely be good to have.

File: nar_common/static/nar_ui/util.js (hunk 3 above)
Comment (jiwalker-usgs) [inline nar_common/static/nar_ui/util.js]: It's up to you, I wouldn't worry about it too much.

**Comment by cschroed-usgs** [comment]
Thanks for the feedback. Will react on Mon.

**Comment by cschroed-usgs** [comment]
Ah, I'm on leave today so by "Mon" I mean "Tues".

File: nar_common/static/nar_ui/util.js (hunk 3 above)
Comment (cschroed-usgs) [inline nar_common/static/nar_ui/util.js]: Roger that. Will skip.

File: nar_common/static/nar_ui/util.js (hunk 4 above)
Comment (cschroed-usgs) [inline nar_common/static/nar_ui/util.js]: Adding a test for it right meow.

File: nar_common/static/nar_ui/util.js (hunk 5 above)
Comment (cschroed-usgs) [inline nar_common/static/nar_ui/util.js]: Ok. I will keep them in the utils class for now. I will keep this idea in mind as I spread changes out to the other pages.

**Comment by cschroed-usgs** [comment]
Merging this and moving on to the detailed graphs.

**Pull request merged**